numpy/lib/function_base.py

```diff
@@ -2400,13 +2400,15 @@ def cov(m, y=None, rowvar=True, bias=False, ddof=None, fweights=None,
     if m.ndim > 2:
         raise ValueError("m has more than 2 dimensions")
 
-    if y is None:
-        dtype = np.result_type(m, np.float64)
-    else:
-        y = np.asarray(y)
-        if y.ndim > 2:
-            raise ValueError("y has more than 2 dimensions")
-        dtype = np.result_type(m, y, np.float64)
+    if dtype is None:
+        dtype = np.float64
+        if y is None:
+            dtype = np.result_type(m, dtype)
+        else:
+            y = np.asarray(y)
+            if y.ndim > 2:
+                raise ValueError("y has more than 2 dimensions")
+            dtype = np.result_type(m, y, dtype)
```

Comment:
You need to leave the `ValueError` outside the `dtype` if
```suggestion
    if y is not None:
        y = np.asarray(y)
        if y.ndim > 2:
            raise ValueError("y has more than 2 dimensions")

    if dtype is None:
        if y is None:
            dtype = np.result_type(m, np.float64)
        else:
            dtype = np.result_type(m, y, np.float64)
```